server/server.js

```diff
@@ -1,8 +1,58 @@
+require('dotenv').config();
 const express = require('express');
 const cors = require('cors');
 const db = require('./db/db-connection.js')
-require('dotenv').config();
 
 const app = express();
 const PORT = process.env.PORT || 1012;
 
+app.use(cors());
+app.use(express.json());
+
+app.get('/', (req, res) => {
+    res.json('You can do hard things, Whitney-Rene')
+})
+
+app.get('/blogpost', async (req, res) => {
+    try {
+        const { rows: blogs } = await db.query('SELECT * FROM blogpost');
+        //After fetching data from the database, this code sends a response to the client
+        res.send(blogs)
+    } catch (error) {
+        return res.status(400).json({error});
+    }
+
+});
+
+app.get('/blogpreview', async (req, res) => {
+    // console.log(res);
+    try{
+        //destructuring assignment syntax  db.query=operation resp for executing sql queryies against db and retunr the result
+        const { rows : blogpreview } = await db.query('SELECT blogpost.blog_picture, blogpost.blog_title, blogpost.blog_publishdate, blogpost.blog_id, authors.author_name, authors.author_id FROM blogpost INNER JOIN authors ON blogpost.author_id = authors.author_id');
+        //result of query is an object, contains properties=one is a rows property, row property contains an array of rows as a result of sql query
+        res.send(blogpreview);
+        // console.log(blogpreview);
+    } catch (error) {
+        return res.status(400).json({ error });
+    }
+});
+
+app.post('/addblogpost', async (req, res) => {
+    //I think I need to insert author first, but I only have one author
+    // CREATE TABLE blogpost (blog_id SERIAL PRIMARY KEY, blog_title VARCHAR (50) NOT NULL, blog_content TEXT NOT NULL, blog_publishdate DATE DEFAULT CURRENT_DATE, blog_picture TEXT, author_id INT REFERENCES authors(author_id));
+    try {
+        const { blog_title, blog_content, blog_picture } = req.body;
+        const result = await db.query(
+            'INSERT INTO blogpost (blog_title, blog_content, blog_picture, author_id) VALUES ($1, $2, $3, $4) RETURNING *',
+            [blog_title, blog_content, blog_picture, 1]
+            
+        );
+        res.status(201).json(result.rows[0]);
+    } catch (error) {
+        return res.status(400).json({error})
+    }
+})
+
+app.listen(PORT, () => {
+    console.log(`Estoy escuchando en ${PORT}`)
+})
```

Comment:
best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

Reply:
Thank you!